### PR TITLE
add memory limits for assembly

### PR DESCRIPTION
**Pull request by SilasK**
The variable `java_mem` and `megahit_memory` were defined in the config file but not passed to all rules. 

I added the memory restriction via the [snakemake key-word resources](http://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#resources) for all rules using scripts of the bbsuite and megahit. 

mem is an integer of the memory allocated for each job in GB. 

The `cluster.json`  can be used to run ATLAS on a slurm cluster with sbatch.

### CODE REVIEW

**Comment by SilasK** [comment]
#20 
a possible solution

File: atlas/rules/assemble.snakefile [hunk 1]
```diff
@@ -79,19 +78,24 @@ rule quality_filter:
...
-        """{SHPFXM} bbduk2.sh {params.inputs} out={output.pe} outs={output.se} \
+        """{SHPFXM} bbduk2.sh {params.inputs} out=good_quality_pe.fastq.gz outs=good_quality_se.fastq.gz \
                {params.rref} {params.lref} mink={params.mink} qout=33 \
                stats={output.stats} hdist={params.hdist} k={params.k} \
                trimq={params.trimq} qtrim={params.qtrim} threads={threads} -Xmx{resources.mem}G \
                minlength={params.minlength} minbasefrequency={params.minbasefrequency} \
-               interleaved={params.interleaved} overwrite=true 2> {log}"""
+               interleaved={params.interleaved} overwrite=true 2> {log}
+
+                cat good_quality_pe.fastq.gz good_quality_se.fastq.gz > {output.pe} 2>> {log}
```
Comment (brwnj) [inline atlas/rules/assemble.snakefile]: This has implications at the normalization and assembly steps. I would recommend implementing read joining after bbduk2.sh then concatenate joined reads, non-joined reads, and quality controlled orphans. The remainder of the protocol would then just be single-end processing.

Comment (SilasK) [reply atlas/rules/assemble.snakefile]: I don't realize what are the implication for bbnorm, but I think the single end reads should also be used to normalize the reads.

**Comment by SilasK** [comment]
@brwnj where do I have to change the `atlas make-config` command, so that the config file generated has the correct values for the memory. (int)

**Comment by brwnj** [comment]
Here it is: https://github.com/pnnl/atlas/blob/master/atlas/conf.py#L114

**Pull request closed without merging**